src/canvas.py

```diff
@@ -75,6 +75,7 @@ def on_draw_event(self, widget, ctx):
             self.figure.draw(self._renderer)
 
     def plot(self, item):
+        self.set_ticks()
```

Comment:
I'd move this in `graphs.refresh()`. That way it gets called just once per refresh not once per item, which should suffice?

Reply:
Yes that definitely suffices. My immediate thinking was that these ticks change when plotting new data, so then they should be refreshed. Putting it in the general refresh may trigger this function in some other cases where it's not needed. But the reverse is also true (putting it in plot also leads to some unnecessary calls).

Putting it in the refresh function may make more sense either way. As it's basically meant to refresh the axes properties.